### PR TITLE
Update noise and weights

### PR DESCRIPTION
**Pull request by Kevin2**
This pull request introduces elevation-dependent noise modeling for visibility simulations, adds support for telescope-specific system temperature corrections, and improves configuration flexibility and data output options. The changes affect configuration files, simulation code, and data structures to enable more realistic noise simulation and support for new output formats.

**Noise modeling and simulation improvements:**

* Added elevation-dependent noise modeling (`noise_mode = "tsys"`) and telescope-specific parameters (currently "meerkat") for system temperature corrections in `src/pyvisgen/simulation/noise.py`. This allows for more realistic noise generation based on antenna elevation and telescope characteristics.
* Updated the `vis_loop` function and related simulation code to support the new noise parameters: `noise_level`, `noise_mode`, and `telescope`. The noise generation now uses the new elevation-dependent model when configured.
* Extended `Baselines` and `ValidBaselineSubset` data structures to include antenna elevation information (`el1`, `el2`), and propagated these through baseline calculation and selection routines. 

**Configuration and output enhancements:**

* Replaced the `noisy` configuration parameter with `noise_level`, added `noise_mode` and `telescope` options, and made related updates in both `config/default_data_set.toml` and `src/pyvisgen/io/config.py` for improved clarity and flexibility.
* Improved writer setup to support "UVH5Writer" as a valid output writer.
* Added a `fits_out_path` option to the `BundleConfig` for optionally writing UVFITS files alongside UVH5 files, including validation to prevent conflicting settings.

**Data handling and type consistency:**

* Changed field of view center coordinates in `SamplingConfig` from `list[int]` to `list[float]` for improved precision.
* Updated visibility data handling to include weights and ensure correct shape and type propagation in FITS writer and visibility structures.

### CODE REVIEW

File: src/pyvisgen/io/config.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ class SamplingConfig(BaseModel, validate_assignment=True):
     seed: str | bool | int | None = 1337
     layout: str = "vlba"
     img_size: int = Field(default=1024, gt=0)
-    fov_center_ra: list[int] = [100, 110]
-    fov_center_dec: list[int] = [30, 40]
+    fov_center_ra: list[float] = [100, 110]
+    fov_center_dec: list[float] = [30, 40]
```
Comment (aknierim) [inline src/pyvisgen/io/config.py:31]: Good catch :)

File: src/pyvisgen/io/datawriters.py [hunk 2]
```diff
@@ -391,6 +398,138 @@ def write(
...
+        │   ├── u
+        │   ├── v
+        │   └── w
+        ├── lmn/
+        │   ├── l
+        │   ├── m
+        │   └── n
+        ├── frequency_bands
+        └── sky/
+            └── SI
```
Comment (aknierim) [inline src/pyvisgen/io/datawriters.py:426]: I think it would be cool if we could also include the station id pairs as metadata here.

Then we could simulate unavailable antennas simply by creating a mask instead of having to simulate the entire dataset all over again:
```python
rng = np.random.default_rng()
unavail = rng.integers(10, size=2)

mask = ~np.isin(st_pairs, unavail).all(axis=1) == True
```
The mask could then even be used in radionets, e.g. in a custom data loader.

I have added the station id pairs in #146 ( https://github.com/radionets-project/pyvisgen/pull/146/commits/5fc11b3ca6f3986086a1b7f07b69518818f483fa), so that all we have to do is save them here, once #146 is merged.

File: src/pyvisgen/io/datawriters.py [hunk 3]
```diff
@@ -391,6 +398,138 @@ def write(
...
+            lmn_grp.create_dataset("l", data=lm[..., 0])
+            lmn_grp.create_dataset("m", data=lm[..., 1])
+            lmn_grp.create_dataset("n", data=n)
+
+            freq_bands = to_numpy(obs.ref_frequency + obs.frequency_offsets)
+            f.create_dataset("frequency_bands", data=freq_bands)
+
+            if sky is not None:
+                sky_grp = f.create_group("sky")
+                sky_grp.create_dataset("SI", data=to_numpy(sky))
```
Comment (aknierim) [inline src/pyvisgen/io/datawriters.py]: Could you move this nested function to a separate method of the class?
That way we avoid re-creating the function every time the `write` method is called.

```suggestion
        lm = self.__to_numpy(obs.lm)  # shape (H, W, 2)
        n = np.sqrt(np.maximum(1.0 - lm[..., 0] ** 2 - lm[..., 1] ** 2, 0.0))

        with File(output_file, "w") as f:
            vis_grp = f.create_group("visibilities")
            vis_grp.create_dataset("V_11", data=self.__to_numpy(vis_data.V_11))
            vis_grp.create_dataset("V_22", data=self.__to_numpy(vis_data.V_22))
            vis_grp.create_dataset("V_12", data=self.__to_numpy(vis_data.V_12))
            vis_grp.create_dataset("V_21", data=self.__to_numpy(vis_data.V_21))
            vis_grp.create_dataset("weights", data=self.__to_numpy(vis_data.weights))

            uvw_grp = f.create_group("uvw")
            uvw_grp.create_dataset("u", data=self.__to_numpy(vis_data.u))
            uvw_grp.create_dataset("v", data=self.__to_numpy(vis_data.v))
            uvw_grp.create_dataset("w", data=self.__to_numpy(vis_data.w))

            lmn_grp = f.create_group("lmn")
            lmn_grp.create_dataset("l", data=lm[..., 0])
            lmn_grp.create_dataset("m", data=lm[..., 1])
            lmn_grp.create_dataset("n", data=n)

            freq_bands = self.__to_numpy(obs.ref_frequency + obs.frequency_offsets)
            f.create_dataset("frequency_bands", data=freq_bands)

            if sky is not None:
                sky_grp = f.create_group("sky")
                sky_grp.create_dataset("SI", data=self.__to_numpy(sky))
                
    def __to_numpy(self, t: torch.Tensor) -> torch.Tensor:
        if isinstance(t, torch.Tensor):
                return t.detach().cpu().numpy()
            return np.asarray(t)
```

File: src/pyvisgen/io/config.py [hunk 4]
```diff
@@ -37,7 +37,10 @@ class SamplingConfig(BaseModel, validate_assignment=True):
     ref_frequency: float = Field(default=15.17600e9, gt=0)
     frequency_offsets: list[float] = [0e8, 1.28e8, 2.56e8, 3.84e8]
     bandwidths: list[float] = [1.28e8, 1.28e8, 1.28e8, 1.28e8]
-    noisy: int = Field(default=0, ge=0)
+    noise_level: float = Field(default=0, ge=0)
+    noise_mode: Literal["sefd", "tsys"] = "sefd"
+    telescope: str = "meerkat"
```
Comment (aknierim) [inline src/pyvisgen/io/config.py]: We could also think about creating a separate `NoiseConfig` for this.

File: src/pyvisgen/simulation/noise.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            ]
+        ),
+        "t_spill_h": torch.tensor(
+            [3.50, 3.20, 3.00, 2.90, 2.80, 2.70, 2.70, 2.75, 2.85, 2.95, 3.05]
+        ),
+        "t_spill_v": torch.tensor(
+            [3.50, 3.20, 3.00, 2.90, 2.80, 2.70, 2.70, 2.75, 2.85, 2.95, 3.05]
+        ),
+    },
+}
```
Comment (aknierim) [inline src/pyvisgen/simulation/noise.py]: Do we want to keep that here or move it inside the layout files?

File: src/pyvisgen/dataset/dataset.py [hunk 6]
```diff
@@ -253,8 +265,13 @@ def _run(self) -> None:
...
             else:
-                for i, vis_data in enumerate(sim_data):
-                    self.writer.write(vis_data, obs, index=i, overwrite=True)
+                for j, vis_data in enumerate(sim_data):
+                    self.writer.write(
+                        vis_data, obs, index=i, sky=SIs[j], overwrite=True
+                    )
+
+                    if fits_writer is not None:
+                        fits_writer.write(vis_data, obs, index=i, overwrite=True)
```
Comment (aknierim) [inline src/pyvisgen/dataset/dataset.py:275]: Isn't that covered by the standard writer already? That would make that `fits_writer` if condition unnecessary...

**Review by aknierim — CHANGES_REQUESTED** [review]

File: src/pyvisgen/io/config.py (hunk 4 above)
Comment (Kevin2) [reply src/pyvisgen/io/config.py]: Good idea.

File: src/pyvisgen/io/datawriters.py (hunk 2 above)
Comment (Kevin2) [reply src/pyvisgen/io/datawriters.py:426]: done :)

File: src/pyvisgen/simulation/noise.py (hunk 5 above)
Comment (Kevin2) [reply src/pyvisgen/simulation/noise.py]: I would suggest extra files in the resources directory. I updated the structure, please have a look.

File: src/pyvisgen/io/datawriters.py (hunk 3 above)
Comment (Kevin2) [reply src/pyvisgen/io/datawriters.py]: done.

File: src/pyvisgen/dataset/dataset.py (hunk 6 above)
Comment (Kevin2) [reply src/pyvisgen/dataset/dataset.py:275]: I think this is necessary for the optional double output in case of the test files.

**Review by aknierim — APPROVED** [review]
